### ISSUE TITLE
when i use custom loss and gpu, fit kernel died

### ISSUE THREAD

**Issue by YooSungHyun** [tensorflow/tensorflow#46417]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): window 10
- TensorFlow version (use command below): tf-nightly-gpu 2.5.0 210112
- Python version:3.8
- CUDA/cuDNN version: 11.1 / maybe 8(?)
- GPU model and memory: RTX 3090 24GB MEM

2. TF 2.0: v1.12.1-48890-g670cc3fa48f 2.5.0-dev20210113

my custom loss function looks like this,

```
import tensorflow as tf
from keras import backend as K
from tensorflow.keras.losses import Loss

@tf.function
def mase(y_true, y_pred, seasonality=1):
    def _naive_forecasting(actual, seasonality: int = 1):
        return actual[:-seasonality]
    
    def _error(actual, predicted):
        return actual - predicted
    
    def _mae(actual, predicted):
        return K.mean(K.abs(_error(actual, predicted)))
    
#     K.print_tensor(y_true,message='\ny_true==')
#     K.print_tensor(y_pred,message='\ny_pred==')
#     K.print_tensor(_mae(y_true, y_pred) / _mae(y_true[seasonality:], _naive_forecasting(y_true, seasonality)),message='\nminus==')
#     print(y_true, y_pred)
    return _mae(y_true, y_pred) / _mae(y_true[seasonality:], _naive_forecasting(y_true, seasonality))

```

and use like this,
```
model.compile(loss=MASE(1), optimizer=Adam(lr=0.001))
...
model.fit(x_concat_data[train], y_concat_data[train], batch_size=batch_size, epochs=epoch, verbose=2, shuffle=True)
...
model.evaluate(x_concat_data[validation], y_concat_data[validation], batch_size=batch_size, callbacks=[early_stopping])
```
I use rtx 3090 and want to train with gpu

cpu training is good!
BUT! when i use gpu, python kernel is dead!
kernel output like this

2021-01-14 09:33:55.296929: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:127] None of the MLIR optimization passes are enabled (registered 2)
Epoch 1/1000
d:/>

what is my problem?

**Comment by YooSungHyun** [comment]
when i use not custom loss func 'mean_absolute_percentage_error' in keras,
i have same issue

**Comment by chazuttu** [comment]
conda install keras 
          or 
Close another notebook which might be running or was used earlier to train a model on GPU

**Comment by YooSungHyun** [comment]
> conda install keras
> or
> Close another notebook which might be running or was used earlier to train a model on GPU

already doin that way. that is not work

**Comment by chazuttu** [comment]
> _Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template_
> 
> **System information**
> 
> * OS Platform and Distribution (e.g., Linux Ubuntu 16.04): window 10
> * TensorFlow version (use command below): tf-nightly-gpu 2.5.0 210112
> * Python version:3.8
> * CUDA/cuDNN version: 11.1 / maybe 8(?)
> * GPU model and memory: RTX 3090 24GB MEM
> 
> 1. TF 2.0: v1.12.1-48890-g670cc3fa48f 2.5.0-dev20210113
> 
> my custom loss function looks like this,
> 
> ```
> import tensorflow as tf
> from keras import backend as K
> from tensorflow.keras.losses import Loss
> 
> @tf.function
> def mase(y_true, y_pred, seasonality=1):
>     def _naive_forecasting(actual, seasonality: int = 1):
>         return actual[:-seasonality]
>     
>     def _error(actual, predicted):
>         return actual - predicted
>     
>     def _mae(actual, predicted):
>         return K.mean(K.abs(_error(actual, predicted)))
>     
> #     K.print_tensor(y_true,message='\ny_true==')
> #     K.print_tensor(y_pred,message='\ny_pred==')
> #     K.print_tensor(_mae(y_true, y_pred) / _mae(y_true[seasonality:], _naive_forecasting(y_true, seasonality)),message='\nminus==')
> #     print(y_true, y_pred)
>     return _mae(y_true, y_pred) / _mae(y_true[seasonality:], _naive_forecasting(y_true, seasonality))
> ```
> 
> and use like this,
> 
> ```
> model.compile(loss=MASE(1), optimizer=Adam(lr=0.001))
> ...
> model.fit(x_concat_data[train], y_concat_data[train], batch_size=batch_size, epochs=epoch, verbose=2, shuffle=True)
> ...
> model.evaluate(x_concat_data[validation], y_concat_data[validation], batch_size=batch_size, callbacks=[early_stopping])
> ```
> 
> I use rtx 3090 and want to train with gpu
> 
> cpu training is good!
> BUT! when i use gpu, python kernel is dead!
> kernel output like this
> 
> 2021-01-14 09:33:55.296929: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:127] None of the MLIR optimization passes are enabled (registered 2)
> Epoch 1/1000
> d:/>
> 
> what is my problem?

2

This informative message is saying MLIR is not being used. Usually its not expected to use the MLIR implementation and are instead expected to use the non-MLIR feature complete implementation.

**Comment by YooSungHyun** [comment]
> > _Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template_
> > **System information**
> > 
> > * OS Platform and Distribution (e.g., Linux Ubuntu 16.04): window 10
> > * TensorFlow version (use command below): tf-nightly-gpu 2.5.0 210112
> > * Python version:3.8
> > * CUDA/cuDNN version: 11.1 / maybe 8(?)
> > * GPU model and memory: RTX 3090 24GB MEM
> > 
> > 
> > 1. TF 2.0: v1.12.1-48890-g670cc3fa48f 2.5.0-dev20210113
> > 
> > my custom loss function looks like this,
> > ```
> > import tensorflow as tf
> > from keras import backend as K
> > from tensorflow.keras.losses import Loss
> > 
> > @tf.function
> > def mase(y_true, y_pred, seasonality=1):
> >     def _naive_forecasting(actual, seasonality: int = 1):
> >         return actual[:-seasonality]
> >     
> >     def _error(actual, predicted):
> >         return actual - predicted
> >     
> >     def _mae(actual, predicted):
> >         return K.mean(K.abs(_error(actual, predicted)))
> >     
> > #     K.print_tensor(y_true,message='\ny_true==')
> > #     K.print_tensor(y_pred,message='\ny_pred==')
> > #     K.print_tensor(_mae(y_true, y_pred) / _mae(y_true[seasonality:], _naive_forecasting(y_true, seasonality)),message='\nminus==')
> > #     print(y_true, y_pred)
> >     return _mae(y_true, y_pred) / _mae(y_true[seasonality:], _naive_forecasting(y_true, seasonality))
> > ```
> > 
> > 
> > and use like this,
> > ```
> > model.compile(loss=MASE(1), optimizer=Adam(lr=0.001))
> > ...
> > model.fit(x_concat_data[train], y_concat_data[train], batch_size=batch_size, epochs=epoch, verbose=2, shuffle=True)
> > ...
> > model.evaluate(x_concat_data[validation], y_concat_data[validation], batch_size=batch_size, callbacks=[early_stopping])
> > ```
> > 
> > 
> > I use rtx 3090 and want to train with gpu
> > cpu training is good!
> > BUT! when i use gpu, python kernel is dead!
> > kernel output like this
> > 2021-01-14 09:33:55.296929: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:127] None of the MLIR optimization passes are enabled (registered 2)
> > Epoch 1/1000
> > d:/>
> > what is my problem?
> 
> 2
> 
> This informative message is saying MLIR is not being used. Usually its not expected to use the MLIR implementation and are instead expected to use the non-MLIR feature complete implementation.

I uninstalled tf-nightly version, and installing tensorflow-gpu 2.4.0.
that version work good

issue happend only tf-nightly version. maybe i think that is nightly version`s bug... but why happened like that.

i will use tensorflow-gpu 2.4.0. so, this issue is complete to me.
